### DISCUSSION TITLE
Formula for approximate size of memory allocation

### DISCUSSION THREAD

**Discussion by kozakromch** [jrouwe/JoltPhysics#1182] [Q&A]
Hello, thanks again for the great library.  It's really comfortable to work with. 
I'm wondering if there is a simple formula that can be used to determine the approximate size of the allocated memory in TempAllocator. I use the standard version.
I know a lot about the physical scene, the exact number of rigid bodies, collisions and so on. And I know that numbers don't change.
Thanks in advance)

**Comment by jrouwe** [comment]
There currently isn't a way to calculate the required amount of memory. The formula would be quite involved and probably also really pessimistic. I would suggest measuring usage and adding some slack.

**Reply by jrouwe** [reply]
Alternatively you could write an implementation that uses a fixed size buffer that is usually sufficient and fall back to malloc if the fixed buffer runs out.

**Reply by jrouwe** [reply]
FYI, since I think it is going to be useful for other people as well, I've implemented a version of this in 3c67be79244fd402dcb0b341cfbca822dafc70dc
